script/makeinstaller.sh

```diff
@@ -1,62 +1,81 @@
 #!/usr/bin/env bash
-# Execute this at the base of the streamlink repo.
+set -e
+
+MAKEINSTALLER=$(basename "$(readlink -f "${0}")")
+log() {
+    echo "[${MAKEINSTALLER}] $@"
+}
+err() {
+    log >&2 "$@"
+    exit 1
+}
+
+
+declare -A DEPS=(
+    [makensis]=NSIS
+    [pynsist]=pynsist
+    [convert]=Imagemagick
+    [inkscape]=inkscape
+    [curl]=curl
+    [jq]=jq
+    [unzip]=unzip
+)
+
+for dep in "${!DEPS[@]}"; do
+    command -v "${dep}" 2>&1 >/dev/null || err "${DEPS["${dep}"]} is required to build the installer. Aborting."
+done
 
-set -e # stop on error
+ROOT=$(git rev-parse --show-toplevel 2>/dev/null || realpath "$(dirname "$(readlink -f "${0}")")/..")
+cd "${ROOT}"
```

Comment:
```suggestion
pushd "${ROOT}"
```

And `popd` on exit :)

Reply:
The entire script is executed in a subshell, so it shouldn't matter whether `cd` or `pushd/popd` is used, right?

Reply:
True, force of habit